api/autopilot.go

```diff
@@ -91,9 +91,35 @@ type (
 		StartTime TimeRFC3339 `json:"startTime"`
 		BuildState
 	}
-)
 
-type (
+	ConfigEvaluationRequest struct {
+		AutopilotConfig    AutopilotConfig    `json:"autopilotConfig"`
+		GougingSettings    GougingSettings    `json:"gougingSettings"`
+		RedundancySettings RedundancySettings `json:"redundancySettings"`
+	}
+
+	ConfigRecommendation struct {
+		GougingSettings GougingSettings `json:"gougingSettings,omitempty"`
+	}
+
+	// ConfigEvaluationResponse is the response type for /evaluate
+	ConfigEvaluationResponse struct {
```

Comment:
The whole struct lacks a bit of structure I think, can we add a `totals` field again? This would also make it more consistent with what we did for `alerts`. I tried writing it out and see why you made it one big struct.
I do think the following is an improvement though:

```go
    ConfigEvaluationResponse struct {
		Hosts    uint64 `json:"hosts"`
		Usable   uint64 `json:"usable"`
		Unusable struct {
			Blocked uint64 `json:"blocked"`
			Gouging struct {
				Contract uint64 `json:"contract"`
				Download uint64 `json:"download"`
				Gouging  uint64 `json:"gouging"`
				Pruning  uint64 `json:"pruning"`
				Upload   uint64 `json:"upload"`
			}
			NotAcceptingContracts uint64 `json:"notAcceptingContracts"`
			NotScanned            uint64 `json:"notScanned"`
			Unknown               uint64 `json:"unknown"`
		} `json:"unusable"`
		Recommendation *ConfigRecommendation `json:"recommendation,omitempty"`
	}
```

- top level props immediately tell you what's happening
- I original had a `totals` breakdown but maybe `unusable` makes more sense
- I think we should keep `unknown` instead of `other`, only because we already expose `unknown` and it maps 1-1, so naming it `other` only introduces a new concept